src/main/java/jenkins/plugins/git/GitSCMTelescope.java

```diff
@@ -212,12 +212,12 @@ public final SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, SCMRevis
                                 getTimestamp(remote, credentials, name)
                         );
                     } else if (name.startsWith(Constants.R_HEADS)) {
-                        head = new SCMHead(name.substring(Constants.R_HEADS.length()));
+                        head = new GitBranchSCMHead(name.substring(Constants.R_HEADS.length()));
                     } else {
                         if (name.startsWith(config.getName() + "/")) {
-                            head = new SCMHead(name.substring(config.getName().length() + 1));
+                            head = new GitBranchSCMHead(name.substring(config.getName().length() + 1));
```

Comment:
I think telescope needs other ref support... we can do that in a separate JIRA as long as there is a TODO referencing it and a release note